simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/StandardExports.java

```diff
@@ -100,69 +98,33 @@ public StandardExports() {
   }
 
   static Long callLongGetter(String getterName, Object obj) throws NoSuchMethodException {
-    Method method = obj.getClass().getMethod(getterName);
-    return (Long) callMethod(method, obj).getValue();
+    return callLongGetter(obj.getClass().getMethod(getterName), obj);
   }
 
   /**
    * Attempts to call a method either directly or via one of the implemented interfaces.
    */
-  static ReturnValue callMethod(Method method, Object obj) {
+  static Long callLongGetter(Method method, Object obj) {
     try {
       method.setAccessible(true);
-      return new ReturnValue(method.invoke(obj));
+      return (Long) method.invoke(obj);
     } catch (Exception e) {
-      LOGGER.log(Level.FINE,
-          "Invocation failed on " + method.getDeclaringClass().getName() + "." + method.getName(),
-          e);
+      // Expected, setAccessible() might fail across Java 9 module boundaries.
     }
 
     for (Class<?> clazz : method.getDeclaringClass().getInterfaces()) {
-      Method interfaceMethod;
       try {
-        interfaceMethod = clazz.getMethod(method.getName(), method.getParameterTypes());
+        Method interfaceMethod = clazz.getMethod(method.getName(), method.getParameterTypes());
+        Long result = callLongGetter(interfaceMethod, obj);
```

Comment:
No. getInterfaces() will return exactly the list of interfaces implemented/extended, in the order they are listed in the class/interface definition. An interface cannot extend itself, so it follows this is a finite sized tree.

Also, we're not really going down all branches. If the method is not declared by one of the implemented interfaces, we don't go down that branch because then that method cannot be declared by any of that interface's ancestors either. I believe there is even a constraint about a method only being defined by a single interface (i.e. you can't implement 2 interfaces with a run() method each), meaning this recursion would only ever result in a simple path traversal.